nanome_rmsd/rmsd_menu.py

```diff
@@ -18,47 +20,160 @@ def _request_refresh(self):
 
     # run the rmsd algorithm
     def _run_rmsd(self):
-        if self._selected_mobile != None or self._selected_target != None:
-            self._plugin.run_rmsd(self._selected_mobile.complex, self._selected_target.complex)
+        # check if both target and receptor are selected
+        if self._selected_mobile == None or self._selected_target == None:
+            self.change_error("unselected",from_run = True)
+            self.make_plugin_usable()
+        # check if one complex is selected as both
+        elif self._selected_mobile.complex.index == self._selected_target.complex.index:
+            self.change_error("select_same",from_run = True)
+            self.make_plugin_usable()
         else:
-            # TODO, need to let the RUN button able to be selected again
-            pass
+            self._plugin.run_rmsd(self._selected_mobile.complex, self._selected_target.complex)
+            self.change_error("unselected",from_run = True)
+            self.change_error("select_same",from_run = True)
+
+
+    # show the error message texts, fromRun means if the it is called after Run is pressed
+    def change_error(self,error_type,from_run = False):
```

Comment:
This function should be split into 2 functions.
One that checks if there are errors.

One that changes the error message. "No error" should be one of the possibilities and it should clear the error message.

The refactored code shouldn't require a "from_run" variable.